### PR TITLE
Junit testing harness

### PR DESCRIPTION
**Pull request by josephhadad05**

### CODE REVIEW

**Review by paulelliott — CHANGES_REQUESTED** [review]
Could we add a single test to the suite just to make sure it actually is executing? It doesn't need to verify anything as long as it has 1 assertion in it.

**Comment by paulelliott** [comment]
There is a commit on this branch, [Merge branch 'main' into JUnit-testing-harness](https://github.com/Creekside-Robotics/Creekside-8775-Robotics/pull/20/commits/c253dc1f10ab7ca058d0ecf25b2167f8537987e3), that shouldn't be here. That's the result of merging the source branch to a target, which is not something you should do.

The main branch is our master record. It is the source of truth for the whole repo. In git terminology, it is _the_ base. When you make a branch off main, you are now on a separate work stream that needs to be kept up to date with main. To do that on your branch, you always want to use a `rebase`, not a `merge`. While you're on your branch you would want to run `git rebase main` and it will replay your changes on top of the new main.

Merging is what you do when you want to bring the code from your branch into main. In that case, from the main branch you would run `git merge JUnit-testing-harness`.

To sum up:
- updating your branch with new changes from main: `git rebase main`
- delivering your changes to main: `git merge your-branch`

Did you do this in the GitHub UI, the cli, or another mechanism?

**Comment by josephhadad05** [comment]
Once I created the pull request github was warning me that there was a conflicting file, and prompted me to solve the conflict. I clicked on the prompt and it gave me an option to edit the file and merge once I was done. This was all done from github.

**Comment by paulelliott** [comment]
OK, I'll look into the project configuration then. Sounds like we have the wrong option checked somewhere.

**Comment by josephhadad05** [comment]
Would you recommend the I go back a commit and rebate the branch correctly?

**Comment by josephhadad05** [comment]
I rebased and added an example test/

**Comment by josephhadad05** [comment]
I rebased and added an example test.

**Review by paulelliott — APPROVED** [review]
This looks great! Well done! :tada:
